### PR TITLE
use simpler batch input

### LINKED ISSUE #251: Batch transactions requires verbose API
The `batch()` transaction API requries you to use the verbose API:

```javascript
await client.batch([
    { sql: "INSERT INTO users (id, name) VALUES (?, ?)", args: [3, "Charlie"] },
    { sql: "INSERT INTO users (id, name) VALUES (?, ?)", args: [4, "Eve"] },
]); 
```

let's add support for the simpler one too:

```javascript
await client.batch([
    ["INSERT INTO users (id, name) VALUES (?, ?)", [4, "Alice"]],
    ["INSERT INTO users (id, name) VALUES (?, ?)", [5, "Bob"]],
]);
```

### PR DESCRIPTION
**Pull request by notrab**
Closes #251

@penberg I didn't yet touch `sqlite3.ts` because updating that might break the compatibility... I'm not sure, wdyt?

Hopefully we can refactor these classes at some point so we don't have to duplicate so much 😅

### CODE REVIEW

**Comment by penberg** [comment]
@notrab I think we need to change `sqlite3.ts` to keep the API compatible across different variants?

**Comment by notrab** [comment]
@penberg you're right, I was thinking this was `libsql-js` for a moment 🤦🏻

File: packages/libsql-client/src/hrana.ts [hunk 1]
```diff
@@ -255,7 +256,7 @@ export async function executeHranaBatch(
     disableForeignKeys: boolean = false,
 ): Promise<Array<ResultSet>> {
     if (disableForeignKeys) {
-        batch.step().run("PRAGMA foreign_keys=off")
+        batch.step().run("PRAGMA foreign_keys=off");
```
Comment (penberg) [inline packages/libsql-client/src/hrana.ts:259]: Diff noise.

File: packages/libsql-client/src/hrana.ts [hunk 2]
```diff
@@ -287,7 +288,7 @@ export async function executeHranaBatch(
         .condition(hrana.BatchCond.not(hrana.BatchCond.ok(commitStep)));
     rollbackStep.run("ROLLBACK").catch((_) => undefined);
     if (disableForeignKeys) {
-        batch.step().run("PRAGMA foreign_keys=on")
+        batch.step().run("PRAGMA foreign_keys=on");
```
Comment (penberg) [inline packages/libsql-client/src/hrana.ts:291]: Diff noise.

File: packages/libsql-client/src/hrana.ts [hunk 3]
```diff
@@ -31,7 +32,7 @@ export abstract class HranaTransaction implements Transaction {
     abstract close(): void;
     abstract get closed(): boolean;
 
-    execute(stmt: InStatement): Promise<ResultSet> {
+    async execute(stmt: InStatement): Promise<ResultSet> {
```
Comment (penberg) [inline packages/libsql-client/src/hrana.ts:35]: Looks unrelated?

File: packages/libsql-client/src/sqlite3.ts [hunk 4]
```diff
@@ -167,9 +170,7 @@ export class Sqlite3Client implements Client {
         }
     }
 
-    async migrate(
-        stmts: Array<InStatement>,
-    ): Promise<Array<ResultSet>> {
+    async migrate(stmts: Array<InStatement>): Promise<Array<ResultSet>> {
```
Comment (penberg) [inline packages/libsql-client/src/sqlite3.ts:173]: Diff noise.

File: packages/libsql-client/src/hrana.ts (hunk 1 above)
Comment (notrab) [reply packages/libsql-client/src/hrana.ts:259]: Prettier ran on commit, have the rules changed?

File: packages/libsql-client/src/sqlite3.ts (hunk 4 above)
Comment (notrab) [reply packages/libsql-client/src/sqlite3.ts:173]: Prettier ran on commit, have the rules changed?

**Comment by DaBigBlob** [comment]
guys is there a reason this isnt merged yet?
